### PR TITLE
QAT model conversion adds extra RESHAPE layers compared to PTQ

### PR DESCRIPTION
**Pull request by SaoirseARM**
Hello,

When converting a QAT FullyConnected model to TFLite, the Reshape layers before and after the FullyConnected layer remain. This is due to the fact that there are Quant/Dequant stat nodes before and after the Reshapes therefore the patterns do not match.
The solution to this is to run  the CreateOptimizePass after the Quantization passes.
Added test to verify this.

Code to reproduce: 
```
import tensorflow as tf
import tensorflow_model_optimization as tfmot

inp = tf.keras.Input(shape=(8, 4), batch_size=1)
out = tf.keras.layers.Dense(16)(inp)
model = tf.keras.Model(inp,out)

def representative_dataset():
    for _ in range(100):
        yield [tf.random.uniform(shape=(1, 8, 4))]

converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.representative_dataset = representative_dataset
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.int8
converter.inference_output_type = tf.int8
tflite_model = converter.convert()
with open('./dense_ptq.tflite', 'wb') as f:
    f.write(tflite_model)

tf.keras.backend.clear_session()

inp = tf.keras.Input(shape=(8, 4), batch_size=1)
out = tfmot.quantization.keras.quantize_annotate_layer(tf.keras.layers.Dense(16))(inp)
model = tfmot.quantization.keras.quantize_apply(tf.keras.Model(inp,out))

converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.inference_input_type = tf.int8
converter.inference_output_type = tf.int8
tflite_model = converter.convert()
with open('./dense_qat.tflite', 'wb') as f:
    f.write(tflite_model)
```
Model produced: 
image-2021-12-08-09-57-17-159

### CODE REVIEW

File: tensorflow/lite/python/lite_v2_test.py [hunk 1]
```diff
@@ -2213,6 +2213,48 @@ def testKerasConvLSTM2DWithMoreThanOneDilationRate(self):
...
+    input_tensor = tf.keras.layers.Input(
+        batch_size=1,
+        shape=[8, 4],
+        name='input_tensor',
+        dtype=tf.float32)
+
+    x = tf.quantization.fake_quant_with_min_max_args(input_tensor, -3.0, 3.0)
+    x = tf.keras.layers.Dense(16)(x)
+    x = tf.quantization.fake_quant_with_min_max_args(x, -3.0, 3.0)
+    model = tf.keras.Model(input_tensor, x)
```
Comment (daverim) [inline tensorflow/lite/python/lite_v2_test.py:2228]: Does this leave a reshape without your changes?

Comment (SaoirseARM) [reply tensorflow/lite/python/lite_v2_test.py:2228]: Hi, 
Thanks for review. Yes, without the change a reshape will remain before and after the FullyConnected layer.
Best Regards,
Saoirse

Comment (daverim) [reply tensorflow/lite/python/lite_v2_test.py:2228]: Looks good to me, but a couple of linting issues:

tensorflow/lite/python/lite_v2_test.py:2370:0: C0301: Line too long (87/80) (line-too-long)
tensorflow/lite/python/lite_v2_test.py:3111:0: C0301: Line too long ([9](https://github.com/tensorflow/tensorflow/pull/55205/checks#step:7:9)6/80) (line-too-long)
tensorflow/lite/python/lite_v2_test.py:3162:0: C0301: Line too long (96/80) (line-too-long)
tensorflow/lite/python/lite_v2_test.py:3188:0: C0301: Line too long (96/80) (line-too-long)

Comment (SaoirseARM) [reply tensorflow/lite/python/lite_v2_test.py:2228]: Thank you. 
I have fixed the issues that were in the new test, but left the ones above as these are not part of the new test. 
I am happy to fix these too, if you would like though.
Best regards,
Saoirse

**Review by daverim — APPROVED** [review]
lgtm

**Comment by gbaned** [comment]
@SaoirseARM Can you please resolve conflicts? Thank you!

**Comment by SaoirseARM** [comment]
> @SaoirseARM Can you please resolve conflicts? Thank you!

Thank you. I have resolved the conflict.
Best regards,
Saoirse

**Review by daverim — APPROVED** [review]
LGTM

**Comment by gbaned** [comment]
@SaoirseARM Can you please address PyLint errors? Thank you!

**Comment by SaoirseARM** [comment]
> @SaoirseARM Can you please address PyLint errors? Thank you!

Thanks, I have now fixed those PyLint errrors.
Best regards, 
Saoirse

**Review by rino20 — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @rino20 Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @rino20 Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
Hi @rino20 Can you please review this PR ? Thank you!

**Comment by rino20** [comment]
> Hi @rino20 Can you please review this PR ? Thank you!

Sorry for late reply, our team member @sngyhan  is reviewing this CL. After approval, we can merge this.

**Review by rino20 — APPROVED** [review]

**Review by sngyhan — APPROVED** [review]
LGTM

**Comment by gbaned** [comment]
@SaoirseARM Can you please resolve conflicts? Thank you!

**Comment by abattery** [comment]
Coinstantaneously, a similar fix has been merged.

**Comment by abattery** [comment]
https://github.com/tensorflow/tensorflow/commit/ad35289082513cddd7cdebeb2a836267c6924ab3

**Pull request closed without merging**